### PR TITLE
[4.0] Frontend: correcting Tags layouts

### PR DESCRIPTION
**Pull request by infograf768**
### Summary of Changes
Following up on similar layouts for Articles List and Contact lists, this one concerns the 3 Tags layouts.
Getting rid of fieldset, separating search filter from list limit. some a11y.
Deleting one redundant lang string, modifying another.
Adding a missing parameter field in xml.

Kept one layout as table, left the others as ul/li.

### Testing Instructions
Create some tags. Tag some contacts and articles.

Create 3 menu items
1. Tagged Items
2. List all tags
3. Compact List of Tagged Items

Display in frontend
patch and test again.

Check code produced
### Actual result BEFORE applying this Pull Request
1. Tagged Items
Screen Shot 2020-10-08 at 17 34 37

2. List all tags
Screen Shot 2020-10-08 at 17 35 03

3. Compact List of Tagged Items
Screen Shot 2020-10-08 at 17 35 40

### Expected result AFTER applying this Pull Request
1. Tagged Items
Screen Shot 2020-10-08 at 17 31 34

2. List all tags
Screen Shot 2020-10-08 at 17 32 47

3. Compact List of Tagged Items
Screen Shot 2020-10-08 at 17 33 13

### CODE REVIEW

**Comment by infograf768** [comment]
### Note
I have remarked some bugs in Tags generally speaking (no surprise...), which are not related to this PR.
Basically the list items and compact list items do not consider an item which `core-state === 0`, neither publishing up or down dates and therefore the possibility for a user who `$cando` only to display them when logged in.

Some items although Published and set to Publish up in the future will display and when a user has no access to that item create a 404 when clicked.

**Comment by Quy** [comment]
The following markup should not be outputted when `Hide` is selected for `Filter Field` and `Display Select`.

```
	<form action="http://localhost/joomla-cms-4.0-dev/index.php/tagged-items" method="post" name="adminForm" id="adminForm">
				
		<input type="hidden" name="filter_order" value="">
		<input type="hidden" name="filter_order_Dir" value="">
		<input type="hidden" name="limitstart" value="">
		<input type="hidden" name="task" value="">
	</form>
```

**Comment by ceford** [comment]
I have tested this item :white_check_mark: successfully on 637291e8afe1f42a8733642f2959eb8e8890e780<br /><br />I think this PR refers only to the Search bar, which did change as indicated in the screen shots. 

I did notice that in menu item creation the Tag field dropdown did not list any tags and if a typed-in tag was spelled wrongly it got created. Separate issues?<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/30998">issues.joomla.org/tracker/joomla-cms/30998</a>.</sub>

**Comment by infograf768** [comment]
> The following markup should not be outputted when `Hide` is selected for `Filter Field` and `Display Select`.

@Quy 
Does not break anything but, if judged important, I guess that depending on the filter display or/and the list limit, these hidden input markup display should be differentiated.
Would you be kind enough to propose mergeable patches ?

> I think this PR refers only to the Search bar, which did change as indicated in the screen shots.

There is also taking of the useless fieldset and some a11y

> I did notice that in menu item creation the Tag field dropdown did not list any tags and if a typed-in tag was spelled wrongly it got created. Separate issues?

Yep, totally separated.

File: components/com_tags/tmpl/tag/default_items.php [hunk 1]
```diff
@@ -29,43 +29,47 @@
 $canCreate    = $user->authorise('core.create', 'com_tags');
 $canEditState = $user->authorise('core.edit.state', 'com_tags');
 ?>
-<form action="<?php echo htmlspecialchars(Uri::getInstance()->toString()); ?>" method="post" name="adminForm" id="adminForm" class="com-tags-tag__items">
-	<?php if ($this->params->get('show_headings') || $this->params->get('filter_field') || $this->params->get('show_pagination_limit')) : ?>
```
Comment (Quy) [inline components/com_tags/tmpl/tag/default_items.php:33]: Add this `if` statement for lines 33-66 should fix the useless markup reported. I am not certain about the `show_headings` parameter.

Comment (infograf768) [reply components/com_tags/tmpl/tag/default_items.php:33]: It is not a table, so `show-headings` was and still is useless.
That parameter is necessary for `tag/list_items.php` which is a table.

But I am confused (and it also concerns other similar layouts):

As it is not a table, I even wonder what
`<input type="hidden" name="filter_order" value="">`
`<input type="hidden" name="filter_order_Dir" value="">`

do here in `tag/default_items.php` as well as in `tags/default_items.php` ?

Comment (infograf768) [reply components/com_tags/tmpl/tag/default_items.php:33]: After discussion, these hidden input are indeed  not necessary. Modifying now.

File: language/en-GB/com_tags.ini [hunk 2]
```diff
@@ -5,10 +5,9 @@
 
 COM_TAGS_CREATED_DATE="Created Date"
 COM_TAGS_DEFAULT_PAGE_TITLE="Tags"
-COM_TAGS_FILTER_SEARCH_DESC="Enter all or part of the title to search for."
 COM_TAGS_MODIFIED_DATE="Modified Date"
 COM_TAGS_NO_ITEMS="No matching items were found."
-COM_TAGS_NO_TAGS="There are no tags."
+COM_TAGS_NO_TAGS="There are no Tags to display."
```
Comment (brianteeman) [inline language/en-GB/com_tags.ini]:
```suggestion
COM_TAGS_NO_TAGS="There are no tags to display."
```

Comment (infograf768) [reply language/en-GB/com_tags.ini]: will do

**Comment by infograf768** [comment]
Corrected useless hidden input for markup + lang string
Left task input although it does not look necessary.

There is no change in display.
Needs another test.

**Comment by paternax** [comment]
I have tested this item :white_check_mark: successfully on 1b3b5dcb777f325e81d7b33bb3c98f3d67fda783<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/30998">issues.joomla.org/tracker/joomla-cms/30998</a>.</sub>

File: components/com_tags/tmpl/tag/default_items.php (hunk 1 above)
Comment (Quy) [reply components/com_tags/tmpl/tag/default_items.php:33]: The `if` should be before the `<form>`.

File: components/com_tags/tmpl/tag/list_items.php [hunk 3]
```diff
@@ -123,5 +121,11 @@
...
 			</div>
 		<?php endif; ?>
-	<?php endif; ?>
-</form>
+		<div>
+			<input type="hidden" name="filter_order" value="<?php echo $listOrder; ?>">
+			<input type="hidden" name="filter_order_Dir" value="<?php echo $listDirn; ?>">
+			<input type="hidden" name="limitstart" value="">
+			<input type="hidden" name="task" value="">
+		</div>
```
Comment (Quy) [inline components/com_tags/tmpl/tag/list_items.php]:
```suggestion
		<input type="hidden" name="filter_order" value="<?php echo $listOrder; ?>">
		<input type="hidden" name="filter_order_Dir" value="<?php echo $listDirn; ?>">
		<input type="hidden" name="limitstart" value="">
		<input type="hidden" name="task" value="">
```

**Comment by infograf768** [comment]
rtc<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/30998">issues.joomla.org/tracker/joomla-cms/30998</a>.</sub>

**Comment by HLeithner** [comment]
The none space between search field and list is expected?

**Comment by infograf768** [comment]
> The none space between search field and list is expected?

Yes, this has been already normalised for articles, contacts, here tags an soon newsfeeds

**Comment by HLeithner** [comment]
That looks horrible mainly because of the rounded corners but maybe the template team came up with something better @richard67 

Thanks anyway

**Comment by infograf768** [comment]
> maybe the template team came up with something better

`horrible ` is a heavy word but, once all is normalised, it will be easier for them.

**Comment by richard67** [comment]
@HLeithner I'm not one of the team leads of the template team.

**Comment by chmst** [comment]
The styling must be made in cassiopeia. It is bad that, if this is made now in jomla cms it looks bad. So maybe we really should wait with merging until the cassiopeia PR is made?

**Comment by richard67** [comment]
@chmst I'll merge it to the Cassiopeia repo in the next hour.

**Comment by richard67** [comment]
Done.

**Comment by AndySDH** [comment]
@infograf768 

Honestly, I'm of the opinion that the Tag Layout should match 1:1 with the Category Layouts.

Meaning these should look the same:

Category Blog = Tagged Items View
Category List = Compact Tagged Items View

I'm wondering why this is not the case?

In Wordpress, all layouts are consistent and identical for all Category view, Tag view, Author view. In Joomla they're all different and inconsistent... Why?

Could Joomla 4 be a good opportunity to uniform these layouts?

**Pull request merged**